cmake/setup_compiler_flags_gnu.cmake

```diff
@@ -36,9 +36,12 @@ ENDIF()
 ########################
 
 #
-# Set -pedantic if the compiler supports it.
+# Set -pedantic if the compiler supports it and if we are not using CUDA. nvcc
+# generates code that produces a lot of warnings when pedantic is enabled. 
 #
-ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-pedantic")
+IF( NOT DEAL_II_WITH_CUDA )
+  ENABLE_IF_SUPPORTED(DEAL_II_CXX_FLAGS "-pedantic")
+ENDIF()
```

Comment:
`setup_compiler_flags` runs before the cuda configuration (and in principle `DEA_II_WITH_CUDA` might change state).
What about simply filtering in `configure_1_cuda.cmake`? Diff:
```
diff --git a/cmake/configure/configure_1_cuda.cmake b/cmake/configure/configure_1_cuda.cmake
index 15e4ae9992..dcff67c648 100644
--- a/cmake/configure/configure_1_cuda.cmake
+++ b/cmake/configure/configure_1_cuda.cmake
@@ -116,6 +116,13 @@ MACRO(FEATURE_CUDA_CONFIGURE_EXTERNAL)
   ADD_FLAGS(DEAL_II_CUDA_FLAGS "${DEAL_II_CXX_VERSION_FLAG}")
 
   #
+  # We cannot use -pedantic as compiler flags. nvcc generates code that
+  # produces a lot of warnings when pedantic is enabled. So filter out the
+  # flag:
+  #
+  STRING(REPLACE "-pedantic" "" DEAL_II_CXX_FLAGS "${DEAL_II_CXX_FLAGS}")
+
+  #
   # Export definitions:
   #
   STRING(SUBSTRING "${CUDA_COMPUTE_CAPABILITY}" 0 1 CUDA_COMPUTE_CAPABILITY_MAJOR)
```